typescript/sdk/src/deploy/proxy.ts

```diff
@@ -135,3 +135,44 @@ export function proxyAdminUpdateTxs(
 
   return transactions;
 }
+
+/**
+ * Check if contract bytecode matches ProxyAdmin patterns
+ * This is more efficient than function calls but less reliable
+ * @param provider The provider to use
+ * @param address The contract address
+ * @returns true if the bytecode suggests it's a ProxyAdmin
+ */
+export async function isProxyAdminByBytecode(
```

Comment:
I'd say that the name sounds better when it is called `isProxyAdminFromBytecode`